### PR TITLE
v6.18.0

### PR DESCRIPTION
**Pull request by alexfauquette**
Noticeable element:

- minor version due to charts becoming stable
- Since no PR refers to the tree view, I removed the section header, and reverteded the version bump proposed by `yarn release:version`.
- Same I revert the proposal of bumping the `@mui/x-license-pro` version proposed by `yarn release:version`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,64 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.18.0
+
+_Nov 3, 2023_
+
+We'd like to offer a big thanks to the 7 contributors who made this release possible. Here are some highlights ✨:
+
+- 🎁 The Charts package is now officially stable!
+- 📈 Line charts now support partial data, and to interpolate missing ones.
```
Comment (LukasTy) [inline CHANGELOG.md:13]:
```suggestion
- 📈 Line charts now support partial data, and can interpolate missing data.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,64 @@
...
+
+#### `@mui/x-data-grid-premium@6.18.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')
+
+Same changes as in `@mui/x-data-grid-pro@6.18.0`.
+
+### Date Pickers
+
+#### `@mui/x-date-pickers@6.18.0`
+
+- [pickers] Add reference links to calendar components  (#10644) @michelengelen
```
Comment (MBilalShafi) [inline CHANGELOG.md:42]:
```suggestion
- [pickers] Add reference links to calendar components (#10644) @michelengelen
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,64 @@
...
+<img width="380" alt="line charts with partial data" src="https://github.com/mui/mui-x/assets/45398769/385ecf77-19b2-4a03-8aef-5d547db1d9ad">
+
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+
+### Data Grid
+
+#### `@mui/x-data-grid@6.18.0`
+
+- [DataGrid] Allow to ignore [diacritics](https://en.wikipedia.org/wiki/Diacritic) when filtering (#10569) @cherniavskii
```
Comment (MBilalShafi) [inline CHANGELOG.md:24]: This enhancement deserves a highlight I think.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,64 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.18.0
+
+_Nov 3, 2023_
+
+We'd like to offer a big thanks to the 7 contributors who made this release possible. Here are some highlights ✨:
+
+- 🎁 The Charts package is now officially stable!
```
Comment (LukasTy) [inline CHANGELOG.md:12]: WDYT about also mentioning the animation support on pie charts? 🤔

**Review by LukasTy — APPROVED** [review]
👏 🎉 🚀

**Review by MBilalShafi — APPROVED** [review]
Congratulations 🥳 on the charts becoming stable. A great official beginning to a really cool package 👏

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,64 @@
...
+- 📈 Line charts now support partial data, and to interpolate missing ones.
+
+<img width="380" alt="line charts with partial data" src="https://github.com/mui/mui-x/assets/45398769/385ecf77-19b2-4a03-8aef-5d547db1d9ad">
+
+- 🐞 Bugfixes
```
Comment (alexfauquette) [inline CHANGELOG.md:17]:
```suggestion
- ✨ Allow to ignore [diacritics](https://en.wikipedia.org/wiki/Diacritic) when filtering
```

File: CHANGELOG.md (hunk 4 above)
Comment (alexfauquette) [inline CHANGELOG.md:12]:
```suggestion
- 🎁 The Charts package is now officially stable!
- 🥧 Pie charts are now animated.
```

**Review by noraleonte — APPROVED** [review]
Awesome release 🎉 🥧

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,65 @@
...
+
+- [charts] Add animation on pie chart (#10782) @alexfauquette
+- [charts] Add reference links to shared/misc chart components (#10660) @michelengelen
+- [charts] Allows to connect nulls (#10803) @alexfauquette
+- [charts] Fix axis highlight in dark mode (#10820) @LukasTy
+
+### Docs
+
+- [docs] Add a recipe for autosizing columns after fetching row-data (#10822) @michelengelen
+- [docs] Add recipe showing how to remove cell focus outline (#10843) @michelengelen
```
Comment (michelengelen) [inline CHANGELOG.md:59]:
```suggestion
- [docs] Add a data grid recipe showing how to remove cell outline on `focus` (#10843) @michelengelen
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,65 @@
...
+### Charts / `@mui/x-charts@6.18.0`
+
+- [charts] Add animation on pie chart (#10782) @alexfauquette
+- [charts] Add reference links to shared/misc chart components (#10660) @michelengelen
+- [charts] Allows to connect nulls (#10803) @alexfauquette
+- [charts] Fix axis highlight in dark mode (#10820) @LukasTy
+
+### Docs
+
+- [docs] Add a recipe for autosizing columns after fetching row-data (#10822) @michelengelen
```
Comment (michelengelen) [inline CHANGELOG.md:58]:
```suggestion
- [docs] Add a data grid recipe for autosizing columns after fetching row-data (#10822) @michelengelen
```

**Review by michelengelen — APPROVED** [review]
just a bit of wording! Awesome release! 🎉 :shipit:

**Review by cherniavskii — APPROVED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,65 @@
...
+
+- [charts] Add animation on pie chart (#10782) @alexfauquette
+- [charts] Add reference links to shared/misc chart components (#10660) @michelengelen
+- [charts] Allows to connect nulls (#10803) @alexfauquette
+- [charts] Fix axis highlight in dark mode (#10820) @LukasTy
+
+### Docs
+
+- [docs] Add a data grid recipe for autosizing columns after fetching row-data (#10822) @michelengelen
+- [docs] Add a data grid recipe showing how to remove cell outline on `focus` (#10843) @michelengelen
```
Comment (LukasTy) [inline CHANGELOG.md:59]:
```suggestion
- [docs] Add a data grid recipe showing how to remove cell outline on `focus` (#10843) @michelengelen
- [docs] Improve custom field input demos readability (#10559) @LukasTy
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,66 @@
...
+- [docs] Improve custom field input demos readability (#10559) @LukasTy
```
Comment (alexfauquette) [inline CHANGELOG.md:60]:
```suggestion
- [docs] Add demo about how to use charts margin (#10886) @alexfauquette
- [docs] Improve custom field input demos readability (#10559) @LukasTy
```

**Pull request merged**